src/main/java/io/fabric8/maven/docker/assembly/DockerAssemblyManager.java

```diff
@@ -237,24 +241,22 @@ void verifyGivenDockerfile(File dockerFile, BuildImageConfiguration buildConfig,
      * Extract all files with a tracking archiver. These can be used to track changes in the filesystem and triggering
      * a rebuild of the image if needed ('docker:watch')
      */
-    public AssemblyFiles getAssemblyFiles(String name, BuildImageConfiguration buildConfig, MojoParameters mojoParams, Logger log)
+    public AssemblyFiles getAssemblyFiles(String name, AssemblyConfiguration assemblyConfig, MojoParameters mojoParams, Logger log)
             throws InvalidAssemblerConfigurationException, ArchiveCreationException, AssemblyFormattingException, MojoExecutionException {
 
         BuildDirs buildDirs = createBuildDirs(name, mojoParams);
 
-        AssemblyConfiguration assemblyConfig = buildConfig.getAssemblyConfiguration();
         String assemblyName = assemblyConfig.getName();
         DockerAssemblyConfigurationSource source =
                         new DockerAssemblyConfigurationSource(mojoParams, buildDirs, assemblyConfig);
         Assembly assembly = getAssemblyConfig(assemblyConfig, source);
 
 
-        synchronized (trackArchiver) {
-            MappingTrackArchiver ta = (MappingTrackArchiver) trackArchiver;
-            ta.init(log, assemblyName);
+        synchronized (trackArchivers) {
+            trackArchivers.init(log, assemblyName);
             assembly.setId("tracker");
             assemblyArchiver.createArchive(assembly, assemblyName, "track", source, false, null);
-            return ta.getAssemblyFiles(mojoParams.getSession());
+            return trackArchivers.getAssemblyFiles(mojoParams.getSession(), assemblyName);
```

Comment:
I think trackArchiver's destFile is no longer being set here after `assemblyArchiver.createArchive(assembly, assemblyName, "track", source, false, null);` call. This seems to be causing NullPointerException in MappingTrackArchiver#getAssemblyFiles() where `getDestFile()` is called. I think explicitly adding destfile should fix this problem:
```suggestion
            File trackArchiverDestFile = assemblyArchiver.createArchive(assembly, assemblyName, "track", source, false, null);
            trackArchivers.get(assemblyName).setDestFile(trackArchiverDestFile);

            return trackArchivers.getAssemblyFiles(mojoParams.getSession(), assemblyName);
```

Reply:
Huh, I must be missing something that was getting set up in the dependency injection, because I can't figure out how this was working before. It's definitely working with this change, though.

Reply:
Is this fix alread applied ? If not, would be cool to add it before merging.

Reply:
I checked and seems like the fix is applied. I'm resolving this comment.